src/workerd/server/server.c++

```diff
@@ -1290,7 +1449,13 @@ kj::Own<Server::Service> Server::makeWorker(kj::StringPtr name, config::Worker::
       kj::atomicRefcounted<IsolateObserver>(),
       name,
       kj::mv(limitEnforcer),
-      false);  // allowInspector -- TODO(beta): support this
+      inspectorService != nullptr);
+
+  // If we are using the inspector, we need to register the Worker::Isolate
+  // with the inspector service.
+  KJ_IF_MAYBE(inspector, inspectorService) {
+    isolate = isolate.attach(inspector->getHandle(name, isolate.get()));
```

Comment:
Two problems here:

* `foo.attach(bar)` arranges for `bar` to be destroyed _after_ `foo`, the idea being that `foo` can depend on `bar` outliving it. Here you would probably prefer the registration be destroyed _before_ the thing it registers. Maybe it doesn't really matter in practice as `~Handle()` doesn't actually access the object but it feels weird.
* `.attach()` is dangerous on refcounted objects because `addRef()` won't carry over the attachment. Maybe in this case it turns out fine because we only care about the main reference holding the registration, but it makes me nervous.

Reply:
Yeah, I had noted that but I really didn't want to make it so that `Worker::Isolate` really knows anything about the `InspectorService` and the registration. What do you think instead about using `WeakIsolateRef` inside `Registration` (`Handle`) instead?

Reply:
Went ahead and updated this to use `WeakIsolateRef`. It turns out the `attach()` had another bug that wasn't immediately obvious in that the handle attachment would be almost immediately destroyed but wasn't cleaning up properly. Using `WeakIsolateRef` means we do the clean up lazily but `Worker::Isolate` does not need to know anything about InspectorService and how to unregister itself.

Reply:
I had been thinking that WorkerService could hold on to the registration, in such a way that it is destroyed before the isolate. But this is fine for now. If we ever add functionality to, say, reload isolates for live services, we may need to do some more work here.

Reply:
I'll add a todo comment